docs/src/main/sphinx/functions/json.rst

```diff
@@ -2,6 +2,1174 @@
+
+.. _json_path_syntax_and_semantics:
+
+JSON path syntax and semantics
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+A JSON path expression, similar to an SQL expression, is a recursive structure.
+Although the name "path" suggests a linear sequence of operations going step by
+step deeper into the JSON structure, a JSON path expression is in fact a tree.
+It can access the input JSON item multiple times, in multiple ways, and combine
+the results. Moreover, the result of a JSON path expression is not a single
+item, but an ordered sequence of items. Each of the sub-expressions takes one
+or more input sequences, and returns a sequence as the result.
+
+.. note::
+
+    In the lax mode, most path operations first unnest all JSON arrays in the
+    input sequence. Any divergence from this rule is mentioned in the following
+    listing. Path modes are explained in :ref:`json_path_modes`.
+
+The JSON path language supports the following features:
+
+literals
+''''''''
+
+- numeric literals
+
+  Their semantics is the same as in SQL.
+
+.. code-block:: text
+
+    -1, 1.2e3, NaN
+
+- string literals
+
+  They are enclosed in double quotes.
+
+.. code-block:: text
+
+    "Some text"
+
+- boolean literals
+
+.. code-block:: text
+
+    true, false
+
+- null literal
+
+  It has the semantics of the JSON null, not of SQL null. See :ref:`json_comparison_rules`.
+
+.. code-block:: text
+
+    null
+
+variables
+'''''''''
+
+- context variable
+
+  It refers to the currently processed input of the JSON
+  function.
+
+.. code-block:: text
+
+    $
+
+- named variable
+
+  It refers to a named parameter by its name.
+
+.. code-block:: text
+
+    $param
+
+- current item variable
+
+  It is used inside the filter expression to refer to the currently processed
+  item from the input sequence.
+
+.. code-block:: text
+
+    @
+
+- last subscript variable
+
+  It refers to the last index of the innermost enclosing array. Array indexes
+  in JSON path expressions are zero-based.
+
+.. code-block:: text
+
+    last
+
+arithmetic binary expressions
+'''''''''''''''''''''''''''''
+
+The JSON path language supports five arithmetic binary operators:
+
+.. code-block:: text
+
+    <path1> + <path2>
+    <path1> - <path2>
+    <path1> * <path2>
+    <path1> / <path2>
+    <path1> % <path2>
+
+Both operands, ``<path1>`` and ``<path2>``, are evaluated to sequences of
+items. For arithmetic binary operators, it is required that both sequences
+consist of singleton numeric items. The arithmetic operation is performed
+accordingly to the SQL semantics, returning a singleton sequence containing
+the result.
+
+The operators follow the usual precedence, and parentheses can be used for
+grouping.
+
+arithmetic unary expressions
+''''''''''''''''''''''''''''
+
+.. code-block:: text
+
+    + <path>
+    - <path>
+
+The operand ``<path>`` is evaluated to a sequence of items. It is required that
+every item is a numeric value. The unary plus or minus is applied to every item
```

Comment:
"... that every item *be* a numeric value".

However, I would rephrase as "Every item must be a numeric value".